catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/MetacardComparator.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.cache.impl;
+
+import ddf.catalog.data.Metacard;
+
+public class MetacardComparator {
+
+    private MetacardComparator() {
+    }
+
+    public static boolean isSame(Metacard cachedMetacard, Metacard latestMetacard) {
+
+        return (cachedMetacard.getAttribute(Metacard.CHECKSUM).toString().equals(latestMetacard.getAttribute(Metacard.CHECKSUM).toString())
+                         &&
+                cachedMetacard.getAttribute(Metacard.MODIFIED)
+                        .toString()
+                        .equals(latestMetacard.getAttribute(Metacard.MODIFIED)
+                                .toString()) &&
+                cachedMetacard.getAttribute(Metacard.RESOURCE_URI)
```

Comment:
Probably a lot cleaner to write a private method that takes two Objects and does the null check/equality check on them.

Then this would be a series of calls like:
```java
//...
&& compare(cachedMetacard.getModifiedDate(), latestMetacard.getModifiedDate())
//...
private boolean compare(Object o1, Object o2) {
    if (o1 == null) {
        return o2 == null;
    }
    return o1.toString().equals(o2.toString());
}
```
That makes that private compare method very similar to a standard `Comparator`s compare method except it

- Doesn't check object assignability because it's a private method and assumed you're only calling it when appropriate
- Doesn't do less-than/greater-than tests because you only care about equality and not order.